### PR TITLE
extraction of the freecoin-lib core components as external lib

### PR DESCRIPTION
**Pull request by jaromil**
renamed all required namespaces also for db operations

extracted source files:

 ── freecoin_lib
    ├── app.clj
    ├── config.clj
    ├── core.clj
    ├── db
    │   ├── account.clj
    │   ├── confirmation.clj
    │   ├── mongo.clj
    │   ├── password_recovery.clj
    │   ├── storage.clj
    │   ├── tag.clj
    │   ├── uuid.clj
    │   └── wallet.clj
    └── utils.clj

pinned dependency from freecoin-lib 0.2 in clojars

### CODE REVIEW

File: test/freecoin/test_helpers/integration.clj [hunk 1]
```diff
@@ -36,7 +37,7 @@
 (defn default-app-config-m []
   (let [stores (s/create-in-memory-stores)]
     {:stores-m stores
-     :blockchain (blockchain/create-in-memory-blockchain :bk)
+     :blockchain (InMemoryBlockchain. :bk (atom {}) (atom {}) (atom {}))
      :config-m {:secure "false"
```
Comment (aspasia) [inline test/freecoin/test_helpers/integration.clj]: this should better remain how it was, by calling the wrapper. Also the import above can be removed

File: test/freecoin/test_helpers/store.clj [hunk 2]
```diff
@@ -1,6 +1,8 @@
 (ns freecoin.test-helpers.store
-  (:import [freecoin.db.mongo MemoryStore]
-           [freecoin.blockchain InMemoryBlockchain]))
+  (:require [freecoin-lib.db.mongo :as db]
+            [freecoin-lib.core :as bc])
+  (:import [freecoin_lib.db.mongo MemoryStore]
+           [freecoin_lib.core InMemoryBlockchain]))
```
Comment (aspasia) [inline test/freecoin/test_helpers/store.clj]: The TestStore is not used anywhere as far as I can tell, can we remove the file?

File: test/freecoin/test_helpers/integration.clj (hunk 1 above)
Comment (jaromil) [inline test/freecoin/test_helpers/integration.clj]: yep! thanks for spotting this, was leftover as a test about the problem you then found how to solve...

File: test/freecoin/test_helpers/store.clj (hunk 2 above)
Comment (jaromil) [inline test/freecoin/test_helpers/store.clj]: removing the file makes the build fail, so I'd rather leave it in for now and proceed

Comment (jaromil) [inline test/freecoin/test_helpers/store.clj]: it is actually used in test/handlers/transactions and sign_in as test-store

**Comment by jaromil** [comment]
all changes addressed excluded the last one since test-store is in use in test/handlers

File: test/freecoin/test_helpers/store.clj (hunk 2 above)
Comment (aspasia) [inline test/freecoin/test_helpers/store.clj]: ok np, just leave it for now, thx

**Pull request merged**